feat: add validation for null values in bookingFieldsResponses

ibex088 opened the pull request:
- Add test case to verify 400 error when bookingFieldsResponses contains null values
- Create ValidateBookingFieldsResponses decorator to reject null values in booking field responses
- Apply validator to CreateBookingInput_2024_08_13.bookingFieldsResponses property
- Ensure all booking field response values are non-null strings

## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)













<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Transform null values in bookingFieldsResponses to empty strings so bookings are accepted and data is normalized. Replaces the custom validator and updates tests to assert the normalization.

- **Bug Fixes**
  - Remove ValidateBookingFieldsResponses and add a Transform on CreateBookingInput_2024_08_13.bookingFieldsResponses to convert null to "".
  - Update e2e test to expect 201 and verify notes and rescheduleReason are saved as empty strings.

<sup>Written for commit 66fd553ef1e280fa86e01b0faea02ac616aa843a. Summary will update automatically on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

ibex088 committed:
feat: add validation for null values in bookingFieldsResponses

- Add test case to verify 400 error when bookingFieldsResponses contains null values
- Create ValidateBookingFieldsResponses decorator to reject null values in booking field responses
- Apply validator to CreateBookingInput_2024_08_13.bookingFieldsResponses property
- Ensure all booking field response values are non-null strings

supalarry reviewed (dismissed):
Looks good! The [Sentry error](https://calcom.sentry.io/issues/7034544127/?referrer=slack&notification_uuid=2e8574e6-215c-488c-b0e3-2aafe66a9ff7&alert_rule_id=15095770&alert_type=issue) is caused by null being passed in booking fields responses and this will prevent it without breaking changes 👍

ThyMinimalDev commented:
please check the impact of such change using Axiom, here is the APL query

```
['api-v2']
| where (['context.requestBody'] contains "bookingFieldsResponses" and ['context.method'] =~ "POST" and ['context.url'] contains "/bookings" and ['context.requestBody'] contains "null")
| extend parsedBody = parse_json(['context.requestBody'])
| extend bookingResponses = parsedBody.bookingFieldsResponses
| where tostring(bookingResponses) contains "null"
| take 50
```

ThyMinimalDev commented on apps/api/v2/src/ee/bookings/2024-08-13/controllers/e2e/booking-fields.e2e-spec.ts:
```diff
@@ -19,12 +19,10 @@ import { UserRepositoryFixture } from "test/fixtures/repository/users.repository
 import { withApiAuth } from "test/utils/withApiAuth";
 
+
+
```
remove plz

ThyMinimalDev commented on apps/api/v2/src/ee/bookings/2024-08-13/controllers/e2e/booking-fields.e2e-spec.ts:
```diff
@@ -1162,6 +1160,24 @@ describe("Bookings Endpoints 2024-08-13", () => {
+        expect(response.body.error.message).toContain(
+          "All values in bookingFieldsResponses must be non-null strings"
+        );
+      });
```
rescheduleReason seems to be a very common field that is often null according to axiom

ibex088 committed:
feat: transform null values to empty strings in bookingFieldsResponses

- Remove ValidateBookingFieldsResponses validator that rejected null values
- Add Transform decorator to convert null values to empty strings in bookingFieldsResponses
- Update test to verify null values are transformed to empty strings instead of returning 400 error

ibex088 committed:
remove extra spaces

ibex088 committed:
test: add rescheduleReason null value test case

ThyMinimalDev committed:
Merge branch 'main' into feat/booking-fields-responses-validator
